src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets

```diff
@@ -40,4 +41,93 @@
       <UseSharedCompilation>true</UseSharedCompilation>
     </PropertyGroup>
   </Target>
+
+  <!--
+    ========================
+    DeterministicSourcePaths
+    ========================
+    
+    Unless specified otherwise enable deterministic source root (PathMap) when building deterministically on CI server, but not for local builds.
+    In order for the debugger to find source files when debugging a locally built binary the PDB must contain original, unmapped local paths.
+  -->
+  <PropertyGroup>
+    <DeterministicSourcePaths Condition="'$(DeterministicSourcePaths)' == '' and '$(Deterministic)' == 'true' and '$(ContinuousIntegrationBuild)' == 'true'">true</DeterministicSourcePaths>
```

Comment:
>ContinuousIntegrationBuild [](start = 117, length = 26)

Btw, it would be helpful to indicate in your documentation for `ContinuousIntegrationBuild` that it's a boolean.
[https://github.com/tmat/repository-info/tree/master/docs](https://github.com/tmat/repository-info/tree/master/docs)

Reply:
Done.